types/gulp-pug/index.d.ts

```diff
@@ -0,0 +1,31 @@
+// Type definitions for gulp-pug 3.3
+// Project: https://github.com/pugjs/gulp-pug#readme
+// Definitions by: remisery <https://github.com/remisery>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+declare function GulpPug(params?: GulpPug.Params): any;
+
+declare namespace GulpPug {
+    interface Params {
```

Comment:
You want to write this as ```interface Params extends Pug.Options``` and omit all the options you copied from that interface. To do this you can import pug as: 
```import * as pug from  'pug'```